In [1]:
import pandas as pd
import spacy
import sys
sys.path.insert(0, r'C:\Users\eduardo.moreno\OneDrive - NEORIS\Desktop\courses\nlp\nlp_nanook\src') # Windows
from utils.utils import clean_text

In [2]:
df_stemm_lemm = pd.read_pickle(r'..\..\..\data\preprocessed\stemm_lemm_stop_words_stop_fb_pages.pkl')

In [3]:
df_stemm_lemm.columns

Index(['Facebook Id', 'Post Created', 'Total Interactions', 'Likes',
       'Comments', 'Shares', 'Love', 'Wow', 'Haha', 'Sad', 'Angry', 'Care',
       'Message', 'Message_stpWrd', 'Message_clean_stpWrd',
       'Message_clean_stemm_stpWrd', 'Message_clean_lemm_stpWrd'],
      dtype='object')

In [4]:
df_stemm_lemm.head()

,Facebook Id,Post Created,Total Interactions,Likes,Comments,Shares,Love,Wow,Haha,Sad,Angry,Care,Message,Message_stpWrd,Message_clean_stpWrd,Message_clean_stemm_stpWrd,Message_clean_lemm_stpWrd
0,97984508920,2014-12-31 23:29:00 CST,13,9,2,2,0,0,0,0,0,0,Energía #nuclear es buena y clave para la solu...,Energía #nuclear clave solución cambio climático,energia nuclear clave solucion cambio climatico,energi nucl clav par solucion cambi climat,energia nuclear clave solucion cambio climatico
1,100050454251406,2014-12-31 22:09:49 CST,44,38,0,6,0,0,0,0,0,0,El cambio climático debería generar la misma a...,cambio climático debería generar atención camb...,cambio climatico deberia generar atencion camb...,cambi climat deberi gener mism atencion cambi ...,cambio climatico deberio generar atencion camb...
2,100064702265343,2014-12-31 18:56:14 CST,7,3,1,3,0,0,0,0,0,0,SIN NOSOTRAS NO HAY REVOLUCIÓN,REVOLUCIÓN,revolucion,nosotr revolucion,revolucion
3,135613897055,2014-12-31 18:00:17 CST,43,37,0,6,0,0,0,0,0,0,"Percy Grandez, especialista en Derecho Ambient...","Percy Grandez, especialista Derecho Ambiental,...",percy grandez especialista derecho ambiental a...,percy grandez especial derech ambiental analiz...,percy grandez especialista derecho ambiental a...
4,260339100808879,2014-12-31 17:33:59 CST,0,0,0,0,0,0,0,0,0,0,Sólo una sexta parte de la cobertura original ...,sexta cobertura original coral permanece Carib...,sexta cobertura original coral permanece carib...,sol sext part cobertur original coral permanec...,sexto cobertura original coral permanecer cari...


In [5]:
nlp = spacy.load('es_core_news_lg')

In [6]:
def lematizar_texto(texto):
    lemas = nlp(texto)
    lematizado = [token.lemma_ for token in lemas]
    return ' '.join(lematizado)

In [7]:
df_stemm_lemm['Message_clean_lemm_stpWrd'][5]

'blog planeta futuro hablar banca etico mujer africano rol ciudad cambio climatico predicción america latino cuento navidad'

In [8]:
lematizar_texto("estaciones")

'estación'

In [9]:
print(f"Lemmatized message: \n{df_stemm_lemm['Message_clean_lemm_stpWrd'][5]}\n")
print(f"Lemmatized + clean message: \n{clean_text(df_stemm_lemm['Message_clean_lemm_stpWrd'][5])}")

Lemmatized message: 
blog planeta futuro hablar banca etico mujer africano rol ciudad cambio climatico predicción america latino cuento navidad

Lemmatized + clean message: 
blog planeta futuro hablar banca etico mujer africano rol ciudad cambio climatico prediccion america latino cuento navidad


In [10]:
clean_lemm = df_stemm_lemm['Message_clean_lemm_stpWrd'].apply(lambda x: clean_text(x))
df_stemm_lemm['Message_clean_lemm_stpWrd'] = clean_lemm

In [11]:
df_stemm_lemm.head()

,Facebook Id,Post Created,Total Interactions,Likes,Comments,Shares,Love,Wow,Haha,Sad,Angry,Care,Message,Message_stpWrd,Message_clean_stpWrd,Message_clean_stemm_stpWrd,Message_clean_lemm_stpWrd
0,97984508920,2014-12-31 23:29:00 CST,13,9,2,2,0,0,0,0,0,0,Energía #nuclear es buena y clave para la solu...,Energía #nuclear clave solución cambio climático,energia nuclear clave solucion cambio climatico,energi nucl clav par solucion cambi climat,energia nuclear clave solucion cambio climatico
1,100050454251406,2014-12-31 22:09:49 CST,44,38,0,6,0,0,0,0,0,0,El cambio climático debería generar la misma a...,cambio climático debería generar atención camb...,cambio climatico deberia generar atencion camb...,cambi climat deberi gener mism atencion cambi ...,cambio climatico deberio generar atencion camb...
2,100064702265343,2014-12-31 18:56:14 CST,7,3,1,3,0,0,0,0,0,0,SIN NOSOTRAS NO HAY REVOLUCIÓN,REVOLUCIÓN,revolucion,nosotr revolucion,revolucion
3,135613897055,2014-12-31 18:00:17 CST,43,37,0,6,0,0,0,0,0,0,"Percy Grandez, especialista en Derecho Ambient...","Percy Grandez, especialista Derecho Ambiental,...",percy grandez especialista derecho ambiental a...,percy grandez especial derech ambiental analiz...,percy grandez especialista derecho ambiental a...
4,260339100808879,2014-12-31 17:33:59 CST,0,0,0,0,0,0,0,0,0,0,Sólo una sexta parte de la cobertura original ...,sexta cobertura original coral permanece Carib...,sexta cobertura original coral permanece carib...,sol sext part cobertur original coral permanec...,sexto cobertura original coral permanecer cari...


Eliminar mensajes duplicados

In [12]:
df_stemm_lemm.shape

(1105612, 17)

In [13]:
df_stemm_lemm.drop_duplicates(subset="Message_clean_lemm_stpWrd").shape

(953693, 17)

In [14]:
1105612-953693

151919

In [15]:
(1105612-953693)/1105612

0.13740715549397076

Hay un total de $151,919$ repetidos, lo cual puede interpretare como "reposts" de otro mensaje.

In [16]:
df_stemm_lemm.to_pickle(r'..\..\..\data\preprocessed\clean_stemm_lemm_stop_words_fb_pages.pkl')

In [17]:
df_stemm_lemm.drop_duplicates(subset="Message_clean_lemm_stpWrd").to_pickle(r'..\..\..\data\preprocessed\clean_stemm_lemm_stop_words_fb_pages_uniques.pkl')